File: src/main/java/net/Chidoziealways/everythingjapanese/villager/ModVillagers.java
package net.Chidoziealways.everythingjapanese.villager;

import com.google.common.collect.ImmutableSet;
import net.Chidoziealways.everythingjapanese.EverythingJapanese;
import net.Chidoziealways.everythingjapanese.block.ModBlocks;
import net.Chidoziealways.everythingjapanese.sound.ModSounds;
import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.ai.village.poi.PoiType;
import net.minecraft.world.entity.npc.VillagerProfession;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModVillagers {
    public static final DeferredRegister<PoiType> POI_TYPES =
            DeferredRegister.create(ForgeRegistries.POI_TYPES, EverythingJapanese.MOD_ID);
    public static final DeferredRegister<VillagerProfession> VILLAGER_PROFESSIONS =
            DeferredRegister.create(ForgeRegistries.VILLAGER_PROFESSIONS, EverythingJapanese.MOD_ID);

    public static final RegistryObject<PoiType> FURNITURE_POI = POI_TYPES.register("furniture_poi",
            () -> new PoiType(ImmutableSet.copyOf(ModBlocks.CHAIR.get().getStateDefinition().getPossibleStates()),
                    1, 1));

    public static final RegistryObject<VillagerProfession> FURNITURE_MAKER = VILLAGER_PROFESSIONS.register("furniture_maker",
            () -> new VillagerProfession(Component.literal("§cFurniture Maker"), holder -> holder.value() == FURNITURE_POI.get(),
                    holder -> holder.value() == FURNITURE_POI.get(), ImmutableSet.of(), ImmutableSet.of(),
                    ModSounds.MAGIC_BLOCK_HIT.get()));

    public static void register(IEventBus eventBus) {
        POI_TYPES.register(eventBus);
        VILLAGER_PROFESSIONS.register(eventBus);
    }
}

File: src/main/kotlin/net/Chidoziealways/everythingjapanese/villager/ModVillagers.kt
package net.Chidoziealways.everythingjapanese.villager

import com.google.common.collect.ImmutableSet
import net.Chidoziealways.everythingjapanese.JAPANESE_MOD_ID
import net.Chidoziealways.everythingjapanese.block.JModBlocks
import net.Chidoziealways.everythingjapanese.sound.ModSounds
import net.minecraft.core.Holder
import net.minecraft.core.registries.Registries
import net.minecraft.network.chat.Component
import net.minecraft.world.entity.ai.village.poi.PoiType
import net.minecraft.world.entity.npc.VillagerProfession
import net.neoforged.bus.api.IEventBus
import net.neoforged.neoforge.registries.DeferredRegister
import java.util.function.Supplier

object ModVillagers {
    val POI_TYPES = DeferredRegister.create(Registries.POINT_OF_INTEREST_TYPE, JAPANESE_MOD_ID)
    val VILLAGER_PROFESSIONS = DeferredRegister.create(Registries.VILLAGER_PROFESSION, JAPANESE_MOD_ID)

    val FURNITURE_POI = POI_TYPES.register(
        "furniture_poi",
        Supplier {
            PoiType(
                ImmutableSet.copyOf(JModBlocks.CHAIR.getStateDefinition().getPossibleStates()),
                1, 1
            )
        })

    val FURNITURE_MAKER = VILLAGER_PROFESSIONS.register(
        "furniture_maker",
        Supplier {
            VillagerProfession(
                Component.literal("§cFurniture Maker"),
                { holder: Holder<PoiType> -> holder.value() === FURNITURE_POI.get() },
                { holder: Holder<PoiType> -> holder.value() === FURNITURE_POI.get() },
                ImmutableSet.of(),
                ImmutableSet.of(),
                ModSounds.MAGIC_BLOCK_HIT.get()
            )
        })

    fun register(eventBus: IEventBus) {
        POI_TYPES.register(eventBus)
        VILLAGER_PROFESSIONS.register(eventBus)
    }
}
